libtransact/src/context/executer_agent.rs

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2019 Bitwise IO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -----------------------------------------------------------------------------
+ */
+
+use std::sync::mpsc::{Receiver, RecvError, Sender};
+
+use crate::context::error::ContextManagerError;
+use crate::context::thread::{
+    ContextOperationMessage, ContextOperationResponse, ContextOperationResult,
+};
+use crate::context::{ContextExecution, ContextId};
+use crate::protocol::receipt::Event;
+
+pub struct ExecuterAgent {
+    tx: Sender<ContextOperationMessage>,
+    rx: Receiver<ContextOperationResponse>,
+}
+
+impl ExecuterAgent {
+    pub fn new(
+        tx: Sender<ContextOperationMessage>,
+        rx: Receiver<ContextOperationResponse>,
+    ) -> Self {
+        ExecuterAgent { tx, rx }
+    }
+}
+
+impl ContextExecution for ExecuterAgent {
+    fn get(
```

Comment:
Shoudl this be get_state() to match the other functions?

Reply:
Do you think this should be changed from the Context Manager up, or just for this external API?